scripts/kc_data_transfer/src/CsvTransfer.py

```diff
@@ -0,0 +1,87 @@
+import csv
+import logging
+from KeyCloakTransfer import FamWrapper
+import constants
+
+LOGGER = logging.getLogger(__name__)
+
+
+class AppToFam:
+    def __init__(self):
+        self.path = constants.APP_USER_FILE_PATH
+        self.fam = FamWrapper()
+
+    def get_forest_client(self, organization: str):
+        # get the forest client number from the "For Organization" column
+        # if has one, it should be in the format of
+        # "[forest_client_number] - [organization_name]"
+        forest_client_num = organization.split('-')[0].replace(" ", "")
+        if forest_client_num.isdigit():
+            return forest_client_num
+        return None
+
+    def get_username_and_type(self, user: str):
+        # get username and type from the "User" column in the csv file
+        # it should be in the format of "USERSTYPE\USERNAME"
+        user_info = user.split('\\')
+        if len(user_info) == 2:
+            if "IDIR" in user_info[0]:
+                return "I", user_info[1]
+            if "BCEID" in user_info[0]:
+                return "B", user_info[1]
+
+        return None, None
+
+    def transfer_user_to_fam(self):
+        app_id = self.fam.get_fom_app_id()
+        users = self.read_file_data(self.path)
+
+        for index, user in enumerate(users):
+            user_type_code, username = self.get_username_and_type(user['User'])
+            if user_type_code and username:
+                role_name = user["Profile"]
+                role_id = self.fam.get_role_id(role_name, app_id)
+
+                if role_id:
+                    print(index, username, user_type_code, role_id,
+                          self.get_forest_client(user["For Organization"]))
+                    self.fam.create_user_role_assignment(
+                        user_type_code=user_type_code,
+                        user_name=username,
+                        role_id=role_id,
+                        role_type_code="C",
+                        forest_client_number=self.get_forest_client(
+                            user["For Organization"]
+                        ),
+                    )
```

Comment:
Just wondering, what happen when `fam.create_user_role_assignment` encounters problem and what we like this csv parsing process to behave in this case? Did this happen before?

Reply:
yes, it happens before. So the script will print out all there users that need to be transferred, and once the script run finished, we compare the number of users that need to be transferred with the number of actually user added in FAM, the FAM frontend shows the total number of users for each application. And if  we feel not confident, we could also log out the status code, that also indicates success or fail
<img width="659" alt="Screen Shot 2023-04-24 at 3 36 36 PM" src="https://user-images.githubusercontent.com/23131735/234131168-5a2fae11-37f2-4d80-ad87-26f001518b22.png">